modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/BaseDataQualifierVisitor.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.introspection;
+
+import org.mule.extension.introspection.DataQualifierVisitor;
+
+/**
+ * Base implementation of {@link DataQualifierVisitor}
+ * in which all methods are implemented and delegate into {@link #defaultOperation()}. In this way,
+ * you can only implement the methods you care about and have a central point to determine which logic
+ * to apply on the cases you don't specifically care about
+ * <p/>
+ * This default implementation of {@link #defaultOperation()} is a no-op
+ *
+ * @since 3.7.0
+ */
+public class BaseDataQualifierVisitor implements DataQualifierVisitor
```

Comment:
BaseDataQualifierVisitor > NullDataQualifierVisitor or AbstractDataQualifierVisitor and make it abstract